rpcperms/interceptor.go

```diff
@@ -124,6 +163,73 @@ func (r *InterceptorChain) SetRPCActive() {
 	defer r.Unlock()
 
 	r.state = rpcActive
+	_ = r.ntfnServer.SendUpdate(r.state)
+}
+
+// SubscribeState subscribes to the current state of the wallet.
+//
+// NOTE: Part of the StateService interface.
+func (r *InterceptorChain) SubscribeState(req *lnrpc.SubscribeStateRequest,
+	stream lnrpc.State_SubscribeStateServer) error {
+
+	sendStateUpdate := func(state rpcState) error {
+
+		resp := &lnrpc.SubscribeStateResponse{}
+		switch state {
+		case walletNotCreated:
+			resp.State = lnrpc.WalletState_NON_EXISTING
+		case walletLocked:
+			resp.State = lnrpc.WalletState_LOCKED
+		case walletUnlocked:
+			resp.State = lnrpc.WalletState_UNLOCKED
+		case rpcActive:
+			resp.State = lnrpc.WalletState_RPC_ACTIVE
+
+		default:
+			return fmt.Errorf("unknown wallet "+
+				"state %v", state)
+
+		}
+
+		return stream.Send(resp)
+	}
+
+	// Subscribe to state updates.
+	client, err := r.ntfnServer.Subscribe()
```

Comment:
I think you need to add `defer client.Cancel()`

Reply:
Good catch!